clusters/app.ci/release-git-sync/release-git-sync-deployment.yaml

```diff
@@ -0,0 +1,53 @@
+kind: List
+apiVersion: v1
+items:
+- apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: release-git-sync
+    namespace: ci
+  spec:
+    replicas: 1
+    strategy:
+      type: RollingUpdate
+    selector:
+      matchLabels:
+        app: release-git-sync
+    template:
+      metadata:
+        labels:
+          app: release-git-sync
+      spec:
+        containers:
+        - name: git-sync
+          command:
+          - /git-sync
+          args:
+          - --repo=https://github.com/openshift/release.git
+          - --branch=master
+          - --wait=30
+          - --root=/tmp/git-sync
+          env:
+          - name: GIT_SYNC_DEST
+            value: release
+          image: gcr.io/google_containers/git-sync:v3.1.6
+          volumeMounts:
+          - name: release
+            mountPath: /tmp/git-sync
+        volumes:
+        - name: release
+          persistentVolumeClaim:
+            claimName: release-git-sync
+- kind: PersistentVolumeClaim
+  apiVersion: v1
+  metadata:
+    labels:
+      app: release-git-sync
+    name: release-git-sync
+    namespace: ci
+  spec:
+    accessModes:
+      - ReadWriteOnce
```

Comment:
In my understanding of persistent storage, this won't the way you describe: _"Every other deployment can just mount this volume instead of multiple config maps."_

Every other volume can only mount the volume **only if they run on the same node** (https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) which will definitely not be the case: you'd need to use ReadWriteMany and that does not seem to be supported in AWS. Also, the consumers really do not need to write.

Why not run this as a sidecar? We do not need this storage to be shared.

Reply:
Even https://github.com/kubernetes/git-sync#git-sync says this is intended to be run in a sidecar.

Reply:
Sidecar is the best option here. I tried to use this approach for better availability. I missed the part that the PV must be in the same node. 

/hold

Reply:
We could still use `ReadWriteMany` but we need a new storage class. I would prefer to go with this approach before going for the sidecar. 

The problem with the sidecar container is that we need 1 init container that first syncs the release repo to the shared volume and 2 containers (component + git-sync). I think it's better to just have a single PV that holds the release repo data rather than syncing the same data for each deployment.

Reply:
Why is having an init container and a sidecar container a problem? That's normal kubernetes stuff. I don't see any advantage of using a shared volume (except for maybe some wasted data and storage): quite the opposite. Single volume creates a single point of failure and increases coupling.

Reply:
I will change the PR to use it as a sidecar. I wanted to make a try to make it faster... it failed miserably :P